src/metrics/abc.rs

```diff
@@ -0,0 +1,803 @@
+
+                // Stops the explorarion when a unary condition is found
+                if matches!(condition_kind, MethodInvocation | Identifier | True | False) {
+                    res += 1.;
+                    break;
+                }
+            }
+
+            if !cursor.goto_next_sibling() {
+                break;
+            }
+        }
+    }
+
+    Some((res, is_bool))
+}
+
+impl Abc for PythonCode {}
+impl Abc for MozjsCode {}
+impl Abc for JavascriptCode {}
+impl Abc for TypescriptCode {}
+impl Abc for TsxCode {}
+impl Abc for RustCode {}
+impl Abc for CppCode {}
+impl Abc for PreprocCode {}
+impl Abc for CcommentCode {}
+
+// Jerry Fitzpatrick (1997) - "Applying the ABC Metric to C, C++, and Java":
+// https://www.softwarerenovation.com/Articles.aspx
+// ABC Java rules: (page 8, figure 4)
+// ABC Java example: (page 15, listing 4)
+impl Abc for JavaCode {
+    fn compute(node: &Node, stats: &mut Stats) {
+        use Java::*;
+
+        match node.object().kind_id().into() {
+            STAREQ | SLASHEQ | PERCENTEQ | DASHEQ | PLUSEQ | LTLTEQ | GTGTEQ | AMPEQ | PIPEEQ
+            | CARETEQ | GTGTGTEQ | PLUSPLUS | DASHDASH => {
+                stats.assignments += 1.;
+            }
+            FieldDeclaration | LocalVariableDeclaration => {
+                stats.declaration.push(DeclKind::Var);
+            }
+            Final => {
+                if let Some(DeclKind::Var) = stats.declaration.last() {
+                    stats.declaration.push(DeclKind::Const);
+                }
+            }
+            SEMI => {
+                if let Some(DeclKind::Const | DeclKind::Var) = stats.declaration.last() {
+                    stats.declaration.clear();
+                }
+            }
+            EQ => {
+                // Excludes constant declarations
+                if let Some(decl) = stats.declaration.last() {
+                    if let DeclKind::Var = decl {
+                        stats.assignments += 1.;
+                    }
+                } else {
+                    stats.assignments += 1.;
+                }
+            }
+            MethodInvocation | New => {
+                stats.branches += 1.;
+            }
+            GTEQ | LTEQ | EQEQ | BANGEQ | Else | Case | Default | QMARK | Try | Catch => {
+                stats.conditions += 1.;
+            }
+            GT | LT => {
+                // Ignores < and > used for generic types
+                if let Some(parent) = node.object().parent() {
+                    if !matches!(parent.kind_id().into(), TypeArguments) {
+                        stats.conditions += 1.;
+                    }
+                }
+            }
+            // Counts unary conditions of elements separated by boolean operators
+            AMPAMP | PIPEPIPE => {
+                if let Some(parent) = node.object().parent() {
+                    if let Some((res, _)) = count_java_unary_conditions(&Node::new(parent)) {
```

Comment:
Instead of using a `res`, can we pass `stats.conditions` as `count_java_unary_conditions` argument and update its value inside that function? In this way we remove a series of duplicated code. Same for all the conditions above